# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: xpcom/base/nsWindowsHelpers.h
# Commit: 7e19c89e818b
# Full Hash: 7e19c89e818b8cf88661386be394e581c8974ab1
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2025-10-06 08:04 +0000
# Regressor Bug: 1980886
# File Overlap Count: 1
# Overlapping Files: xpcom/io/nsLocalFileWin.cpp
# Description:
#   Bug 1980886 p3 - Ensure no old inherited ACEs remain when moving files on Windows. r=yjuglaret
# ==============================================================================

diff -r e1b091b14517 -r 7e19c89e818b xpcom/base/nsWindowsHelpers.h
--- a/xpcom/base/nsWindowsHelpers.h	Mon Oct 06 08:04:25 2025 +0000
+++ b/xpcom/base/nsWindowsHelpers.h	Mon Oct 06 08:04:25 2025 +0000
@@ -328,6 +328,19 @@
   }
 };
 
+using AutoFreeSecurityDescriptor =
+    mozilla::UniquePtr<SECURITY_DESCRIPTOR, LocalFreeDeleter>;
+
+struct DestroyPrivateObjectSecurityDeleter {
+  void operator()(PSECURITY_DESCRIPTOR aSecDescPtr) {
+    ::DestroyPrivateObjectSecurity(&aSecDescPtr);
+  }
+};
+
+using AutoDestroySecurityDescriptor =
+    mozilla::UniquePtr<SECURITY_DESCRIPTOR,
+                       DestroyPrivateObjectSecurityDeleter>;
+
 // One caller of this function is early in startup and several others are not,
 // so they have different ways of determining the two parameters. This function
 // exists just so any future code that needs to determine whether the dynamic